Quickstart/azure-search-quickstart/azure-search-quickstart/Program.cs

```diff
@@ -0,0 +1,267 @@
+                    Description = "Sublime Cliff Hotel is located in the heart of the historic center of Sublime in an extremely vibrant and lively area within short walking distance to the sites and landmarks of the city and is surrounded by the extraordinary beauty of churches, buildings, shops and monuments. Sublime Cliff is part of a lovingly restored 1800 palace.",
+                    DescriptionFr = "Le sublime Cliff Hotel est situé au coeur du centre historique de sublime dans un quartier extrêmement animé et vivant, à courte distance de marche des sites et monuments de la ville et est entouré par l'extraordinaire beauté des églises, des bâtiments, des commerces et Monuments. Sublime Cliff fait partie d'un Palace 1800 restauré avec amour.",
+                    Category = "Boutique",
+                    Tags = new[] { "concierge", "view", "24-hour front desk service" },
+                    ParkingIncluded = true,
+                    LastRenovationDate = new DateTimeOffset(1960, 2, 06, 0, 0, 0, TimeSpan.Zero),
+                    Rating = 4.6,
+                    Address = new Address()
+                    {
+                        StreetAddress = "7400 San Pedro Ave",
+                        City = "San Antonio",
+                        StateProvince = "TX",
+                        PostalCode = "78216",
+                        Country = "USA"
+                    }
+                }
+            ),
+         };
+
+            var batch = IndexBatch.New(actions);
+
+            try
+            {
+                indexClient.Documents.Index(batch);
+            }
+            catch (IndexBatchException e)
+            {
+                // When a service is under load, indexing might fail for some documents in the batch. 
+                // Depending on your application, you can compensate by delaying and retrying. 
+                // For this simple demo, we just log the failed document keys and continue.
+                Console.WriteLine("Failed to index some of the documents: {0}",
+                    String.Join(", ", e.IndexingResults.Where(r => !r.Succeeded).Select(r => r.Key)));
+            }
+
+            // Wait 2 seconds before starting queries
+            Console.WriteLine("Waiting for indexing...\n");
+            Thread.Sleep(2000);
+        }
+
+        // Add query logic and handle results
+        private static void RunQueries(ISearchIndexClient indexClient)
+        {
+            SearchParameters parameters;
+            DocumentSearchResult<Hotel> results;
+
+            // Query 1 
+            Console.WriteLine("Query 1: Search for term 'Atlanta' with no result trimming");
+            parameters = new SearchParameters();
+            results = indexClient.Documents.Search<Hotel>("Atlanta", parameters);
+            WriteDocuments(results);
+
+
+            // Query 2
+            Console.WriteLine("Query 2: Search on the term 'Atlanta', with trimming");
```

Comment:
It seems like there's supposed to be a filter here, but I don't see one. Also, it wouldn't be a great showcase for filters since if I'm not mistaken "Atlanta" should only yield one result.

Reply:
Ok, I see where I went wrong -- "trimming" is being used to mean "projection" or "selection". That's really confusing. Terminology is important!

Reply:
I fixed the terminology -- it's about the results (full | partial)